app/assets/javascripts/dashboard.js

```diff
@@ -82,19 +97,20 @@ function serialize_grid(){
     return result;
 }
 
-
 function fill_restore_list(){
    $("ul>li.widget-restore").remove();
    var restore_list = [];
    var hidden_widgets = $(".gridster>ul>li[data-hide='true']");
    if (hidden_widgets.exists()){
        hidden_widgets.each(function(i, widget) {
-           restore_list.push( "<li class='widget-restore'><a href='#' onclick='show_widget(\""+$(widget).attr('data-id')+"\")'>" + $(widget).attr('data-name') + "</a></li>" );
+           restore_list.push("<li class='widget-restore'><a href='#' onclick='show_widget(\"" +
+               $(widget).attr('data-id') + "\")'>" +
+               $(widget).attr('data-name') + "</a></li>");
        });
    } else {
        restore_list.push("<li class='widget-restore'><a>" + __('Nothing to restore') + "</a></li>");
    }
-   $('#restore_list').parent('ul').append(restore_list.join(" "));
+   $('#restore_list').after(restore_list.join(" "));
```

Comment:
This ensures we put the hidden widgets in the restore list instead of the 'Add widgets' list.